run.py

```diff
@@ -0,0 +1,317 @@
+    """
+    # make sure we only send max number of entires per bibcode/source_file to be queried
+    if bibcodes:
+        bibcodes = bibcodes[:MAX_ENTRIES_DIAGNOSTICS]
+    if source_filenames:
+        source_filenames = source_filenames[:MAX_ENTRIES_DIAGNOSTICS]
+    results = app.query_reference_tbl(bibcodes, source_filenames)
+    for result in results:
+        print(result)
+    return
+
+
+def get_source_filenames(source_file_path, file_extension, date_cutoff):
+    """
+    :param source_file_path:
+    :param date_cutoff: if modified date is after this date
+    :return: list of files in the directory with modified date after the cutoff, if any
+    """
+    list_files = []
+    for root, dirs, files in os.walk(source_file_path):
+        for basename in files:
+            if fnmatch.fnmatch(basename, file_extension):
+                filename = os.path.join(root, basename)
+                if get_date_modified_struct_time(filename) >= date_cutoff:
+                    list_files.append(filename)
+    return list_files
+
+
+def check_queue(queued_tasks):
+    """
+    every few seconds check the tasks to see if it was processed
+
+    :param queued_tasks:
+    :return:
+    """
+    def output_message(message):
+        """
+
+        :param message:
+        :return:
+        """
+        logger.info(message)
+        print(message)
+
+
+    to_queue_tasks = []
+    # check to see when/if the queued tasks got processed successfully
+    for queued_task in queued_tasks:
+        if queued_task['results'].state == 'SUCCESS' and queued_task['results'].result:
+            if 'filename' in queued_task:
+                output_message("Reference file %s successfully processed."%queued_task['filename'])
+            elif 'reprocess' in queued_task:
+                record = queued_task['reprocess']
+                output_message("Reprocessed %d reference(s) from source file %s successfully processed." %(len(record['references']), record['source_filename']))
+        else:
+            # queue the task to check it again in few seconds, if it has not been too long
+            if queued_task['results'].date_done:
+                if (queued_task['results'].date_done - datetime.now()).seconds > config['TASK_PROCESS_TIME']:
+                    output_message("Reference file %s was not processed in allotted time! Removed from verification queue." % queued_task['filename'])
+                    continue
+            to_queue_tasks.append(queued_task)
+    if len(to_queue_tasks) > 0:
+        output_message('%d/%d task remains for processing.'%(len(to_queue_tasks),len(queued_tasks)))
+        threading.Timer(config['QUEUE_AUDIT_INTERVAL'], check_queue, (to_queue_tasks,)).start()
+    else:
+        output_message('All tasks consumed.')
+
+
+def queue_files(filenames):
+    """
+    queue all the requested files
+
+    :param files:
+    :return:
+    """
+    queued_tasks = []
+    for filename in filenames:
```

Comment:
I like the changes; I'd advocate for going even farther (sending apples for processing, not crates of apples):

please consider this pattern:

```
for filename in filenames:
   parser = parser(app.get_parser_name(filename))
   for bibcode, raw_reference in parser.parse_input(filename):
       tasks.tasks_process_references.apply_async({'bibcode': bibcode, 'raw': reference_string})
```

also careful about the `check_queue` logic above; calling `tasks.task_x.delay` without parameters will submit the message to the queue - so that you can verify its status (if you keep the message response object). So all the code accomplishes is to say 'during the time I was willing to wait, the messages were (not) processed' -- but they can of course be processed at some time afterwards. The allotted time `TASK_PROCESS_TIME` will not affect task.

i guess it's a long winded way of saying: the same results could be had by just monitoring queue length; or checking messages in loop and sleeping -- it's really not clear why the async timer -- perhaps you wanted to call submission multiple times?

I personally don't derive much value from having to wait; tasks are already submitted - `run.py` could exit. The `check_queue` is only going to say what logs is already saying (and it adds to memory/cpu requirements) but other than that it is not going to guarantee that it exits *after all* tasks were processed.